src/constituents.cc

```diff
@@ -0,0 +1,224 @@
+#include <fnmatch.h>
+#include <nlohmann/json.hpp>
+#include <nix/config.h>
+#include <nix/derivations.hh>
+#include <nix/local-fs-store.hh>
+
+#include "constituents.hh"
+
+namespace {
+// This is copied from `libutil/topo-sort.hh` in CppNix and slightly modified.
+// However, I needed a way to use strings as identifiers to sort, but still be
+// able to put AggregateJob objects into this function since I'd rather not have
+// to transform back and forth between a list of strings and AggregateJobs in
+// resolveNamedConstituents.
```

Comment:
Is this an explanation for why you didn't just use `libutil/topo-sort.hh`?

Reply:
I feel like if instead of having 

`std::set<AggregateJob>` you had `std::map<std::string, AggregateJob>` you could both

1. Use `libutil/topo-sort.hh` as-is

2. Use the automatic `auto operator<=>(const AggregateJob &) const;` rather than one which is "not lawful"

3. The final `std::vector<std::string>` to `std::vector<std::pair<std::string, AggregateJob>>` would be not too bad, either a simple or `std::transform`?

Reply:
I tried to do this in https://github.com/nix-community/nix-eval-jobs/pull/364/commits/d9febaf16ba72a47a9be4ac9152cf36ef16ba4c2 but it seems to compute the wrong result.
So reverted the commit again. However feel free to fix this if you see what is wrong. To me it seems these both sorting functions are not functionally equivalent.